src/rust/ide/lib/span-tree/src/generate.rs

```diff
@@ -207,17 +213,19 @@ impl SpanTreeGenerator for Ast {
 
 // === Operators (Sections and Infixes) ===
 
-impl SpanTreeGenerator for ast::opr::Chain {
-    fn generate_node(&self, kind:node::Kind, context:&impl Context) -> FallibleResult<Node> {
+impl<T:Payload> SpanTreeGenerator<T> for ast::opr::Chain {
+    fn generate_node
+    (&self, kind:impl Into<node::Kind>, context:&impl Context) -> FallibleResult<Node<T>> {
         // Removing operands is possible only when chain has at least 3 of them
         // (target and two arguments).
-        let is_removable                                 = self.args.len() >= 2;
-        let node_and_offset:FallibleResult<(Node,usize)> = match &self.target {
+        let kind      = kind.into();
+        let removable = self.args.len() >= 2;
+        let node_and_offset:FallibleResult<(Node<T>,usize)> = match &self.target {
             Some(target) => {
-                let node = target.arg.generate_node(node::Kind::Target {is_removable},context)?;
+                let node = target.arg.generate_node(node::Kind::this().removable(),context)?;
```

Comment:
Tests are failing, and here is the (one of) reason(s): before we mark infix arguments as removable only when there were more than 2 of them.